platforms/android/aar-creating/build_java_shared_aar.py

```diff
@@ -0,0 +1,92 @@
+import argparse
+from os import path
+import os
+import shutil
+import subprocess
+
+from build_static_aar import fill_template
+
+
+COPY_FROM_SDK_TO_ANDROID_PROJECT = [
+    ["sdk/native/jni/include", "OpenCV/src/main/cpp/include"],
+    ["sdk/java/src/org", "OpenCV/src/main/java/org"],
+    ["sdk/java/res", "OpenCV/src/main/res"]
+]
+
+ABIS = ["arm64-v8a", "armeabi-v7a", "x86", "x86_64"] # if you want to change it, you also need to change Android project template
+
+COPY_FROM_SDK_TO_APK = [
+    ["sdk/native/libs/<ABI>/libopencv_java4.so", "jni/<ABI>/libopencv_java4.so"],
+    ["sdk/native/libs/<ABI>/libopencv_java4.so", "prefab/modules/opencv_java4/libs/android.<ABI>/libopencv_java4.so"],
+]
+
+ANDROID_PROJECT_TEMPLATE_DIR = "OpenCVAndroidProject"
+ANDROID_PROJECT_DIR = "build_java_shared/AndroidProject"
+COMPILED_AAR_PATH = path.join(ANDROID_PROJECT_DIR, "OpenCV/build/outputs/aar/opencv-release.aar")
+AAR_UNZIPPED_DIR = "build_java_shared/aar_unzipped"
+FINAL_AAR_PATH = "outputs/opencv_java_shared.aar"
+FINAL_REPO_PATH = "outputs/maven_repo"
+
+def main(sdk_dir, opencv_version):
+    print("Preparing Android project...")
+    # ANDROID_PROJECT_TEMPLATE_DIR contains an Android project template that creates AAR
+    shutil.copytree(ANDROID_PROJECT_TEMPLATE_DIR, ANDROID_PROJECT_DIR)
+
+    # Configuring the Android project to Java + shared C++ lib version
+    shutil.rmtree(path.join(ANDROID_PROJECT_DIR, "OpenCV/src/main/cpp/include"))
+
+    fill_template(path.join(ANDROID_PROJECT_DIR, "OpenCV/build.gradle.template"),
+                  path.join(ANDROID_PROJECT_DIR, "OpenCV/build.gradle"),
+                  {"LIB_NAME": "opencv_java4", "LIB_TYPE": "c++_shared", "PACKAGE_NAME": "opencv", "OPENCV_VERSION": opencv_version})
+    fill_template(path.join(ANDROID_PROJECT_DIR, "OpenCV/src/main/cpp/CMakeLists.txt.template"),
+                  path.join(ANDROID_PROJECT_DIR, "OpenCV/src/main/cpp/CMakeLists.txt"),
+                  {"LIB_NAME": "opencv_java4", "LIB_TYPE": "SHARED"})
+
+    # Copying Java code and C++ public headers from SDK to the Android project
+    for src, dst in COPY_FROM_SDK_TO_ANDROID_PROJECT:
+        shutil.copytree(path.join(sdk_dir, src),
+                        path.join(ANDROID_PROJECT_DIR, dst))
+
+    print("Running gradle assembleRelease...")
+    # Running gradle to build the Android project
+    subprocess.run(["gradlew", "assembleRelease"],
+                shell=True,
+                cwd=ANDROID_PROJECT_DIR)
```

Comment:
Please check build status. I get error on the next steps, if something went wrong:
```
Adding libs to AAR...
Traceback (most recent call last):
  File "./build_java_shared_aar.py", line 92, in <module>
    main(args.opencv_sdk_path, args.opencv_version)
  File "./build_java_shared_aar.py", line 60, in main
    shutil.unpack_archive(COMPILED_AAR_PATH, AAR_UNZIPPED_DIR, "zip")
  File "/usr/lib/python3.6/shutil.py", line 974, in unpack_archive
    func(filename, extract_dir, **dict(format_info[2]))
  File "/usr/lib/python3.6/shutil.py", line 883, in _unpack_zipfile
    raise ReadError("%s is not a zip file" % filename)
shutil.ReadError: build_java_shared/AndroidProject/OpenCV/build/outputs/aar/opencv-release.aar is not a zip file
```

Reply:
I probably fixed the problem, please try again